src/server.cpp

```diff
@@ -37,44 +35,33 @@ void Server::handle_receive(const error_code& error, std::size_t bytes)
         return;
     }
 
-    size_t length = recv_buffer_[2] * 256 + recv_buffer_[3];
-
-    if (bytes < length)
+    try
     {
-        std::cout << "Error: request length " << bytes << "is less than specified in the request - " << length << "\n";
-        return;
-    }
-
-    if (recv_buffer_[0] == ACCESS_REQUEST)
-        send_buffer_[0] = ACCESS_ACCEPT;
-    else
-        send_buffer_[0] = ACCESS_REJECT;
-
-    send_buffer_[1] = recv_buffer_[1];
-    send_buffer_[2] = 0;
-    send_buffer_[3] = 20;
-
-    for (size_t i = 0; i < 16; ++i)
-        send_buffer_[i + 4] = recv_buffer_[i + 4];
-
-    std::string secr("secret");
-
-    for (size_t i = 0; i < 6; ++i)
-        send_buffer_[i + 20] = secr[i];
-
-    std::array<uint8_t, 16> md;
+        Packet packet = makeResponse(Packet(m_recvBuffer, bytes));
 
-    MD5(send_buffer_.data(), 20 + secr.length(), md.data());
+        m_socket.async_send_to(boost::asio::buffer(packet.makeSendBuffer("secret")),
+            m_remoteEndpoint, std::bind(&Server::handleSend, this, pls::_1, pls::_2));
 
-    for (size_t i = 0; i < 16; ++i)
-        send_buffer_[i + 4] = md[i];
-
-    socket_.async_send_to(boost::asio::buffer(send_buffer_, 20), remote_endpoint_,
-        std::bind(&Server::handle_send, this, pls::_1, pls::_2));
+        startReceive();
+    }
+    catch (const std::runtime_error& exception)
+    {
+        std::cout << "Runtime error: " << exception.what() << "\n";
```

Comment:
'Runtime error' does not report any usefull information. If you have nothing to say except `exception.what()` then sey just `exception.what()`. No need to tell everybody that this is a runtime exception, it's obvious.